code/modules/vehicles/cars/clowncar.dm

```diff
@@ -93,111 +95,129 @@
 	initialize_controller_action_type(/datum/action/vehicle/sealed/cannon, VEHICLE_CONTROL_DRIVE)
 	AddElement(/datum/element/waddling)
 
-/obj/vehicle/sealed/car/clowncar/Destroy()
+/obj/vehicle/sealed/car/clowncar/obj_destruction(damage_flag)
 	playsound(src, 'sound/vehicles/clowncar_fart.ogg', 100)
 	return ..()
 
-/obj/vehicle/sealed/car/clowncar/Move(newloc, dir)
-	. = ..()
-	if(droppingoil)
-		new /obj/effect/decal/cleanable/oil/slippery(loc)
-
-/obj/vehicle/sealed/car/clowncar/proc/RollTheDice(mob/user)
-	if(world.time - lastRTDtime < RTDcooldown)
+/**
+ * Plays a random funky effect
+ * Only available while car is emagged
+ * Possible effects:
+ * * Spawn bananapeel
+ * * Spawn random reagent foam
+ * * Make the clown car look like a singulo temporarily
+ * * Spawn Laughing chem gas
+ * * Drop oil
+ * * Fart and make everyone nearby laugh
+ */
+/obj/vehicle/sealed/car/clowncar/proc/roll_the_dice(mob/user)
+	if(TIMER_COOLDOWN_CHECK(src, COOLDOWN_CLOWNCAR_RANDOMNESS))
```

Comment:
Another instance of "why is this a timer"

Reply:
ditto see above